cli/command/image/build.go

```diff
@@ -477,3 +479,25 @@ func replaceDockerfileTarWrapper(ctx context.Context, inputTarStream io.ReadClos
 
 	return pipeReader
 }
+
+func createBuildArgs(cfg *configfile.ConfigFile, o opts.ListOpts) map[string]*string {
+	permitted := map[string]*string{
+		"HTTP_PROXY":  &cfg.HTTPProxy,
+		"HTTPS_PROXY": &cfg.HTTPSProxy,
+		"NO_PROXY":    &cfg.NoProxy,
+		"FTP_PROXY":   &cfg.FTPProxy,
+	}
+	m := runconfigopts.ConvertKVStringsToMapWithNil(o.GetAll())
+	for k := range permitted {
+		if *permitted[k] == "" {
+			continue
+		}
+		if _, ok := m[k]; !ok {
+			m[k] = permitted[k]
+		}
+		if _, ok := m[strings.ToLower(k)]; !ok {
```

Comment:
interesting, didn't notice before that we actually list both variants in the code.